src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackProcessor.java

```diff
@@ -96,13 +112,24 @@ private void attachCase(String exceptionRecordJurisdiction,
         return emptyList();
     }
 
-    @SuppressWarnings({"unchecked", "squid:S1135"})
-    //TODO WIP
     private Map<String, Object> insertNewScannedDocument(List<Map<String, Object>> exceptionDocuments,
-                                                         Map<String, Object> caseData) {
-        //TODO check that document Id is unique and not duplicate in caseData
-        List<Object> caseList = (List<Object>) caseData.get(SCANNED_DOCUMENTS);
-        caseList.addAll(exceptionDocuments);
-        return ImmutableMap.of(SCANNED_DOCUMENTS, caseList);
+                                                         List<Map<String, Object>> existingDocuments) {
+        //TODO assert that there is one document in the exception record.
+        existingDocuments.add(exceptionDocuments.get(0));
+        return ImmutableMap.of(SCANNED_DOCUMENTS, existingDocuments);
+    }
+
+    private void checkForDuplicateAttachment(List<Map<String, Object>> exceptionDocuments,
+                                             String caseRef,
+                                             List<Map<String, Object>> existingDocuments) {
+        String exceptionRecordId = (String) exceptionDocuments.get(0).get("documentNumber");
```

Comment:
always only 1?

Reply:
yup as mentioned above...
I've only seen one so and there are no real requirements written down.
Thats why we have a PR and the //TODO check assumption :smile:

Reply:
updated the TODO and will deal with more than one doc in the next PR